File: mobile/src/main/java/technology/mainthread/apps/gatekeeper/model/event/GatekeeperState.java
package technology.mainthread.apps.gatekeeper.model.event;

import android.support.annotation.StringDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.SOURCE)
@StringDef({GatekeeperState.ONLINE, GatekeeperState.OFFLINE, GatekeeperState.PRIMED,
        GatekeeperState.DOOR_OPEN, GatekeeperState.UNKNOWN})
public @interface GatekeeperState {
    String ONLINE = "ONLINE";
    String OFFLINE = "OFFLINE";
    String DOOR_OPEN = "DOOR_OPEN";
    String PRIMED = "PRIMED";
    String UNKNOWN = "UNKNOWN";
}

File: mobile/src/main/java/technology/mainthread/apps/gatekeeper/model/event/GatekeeperState.kt
package technology.mainthread.apps.gatekeeper.model.event

enum class GatekeeperState(val label: String) {
    ONLINE("ONLINE"),
    OFFLINE("OFFLINE"),
    DOOR_OPEN("DOOR_OPEN"),
    PRIMED("PRIMED"),
    UNKNOWN("UNKNOWN")
}